velox/functions/sparksql/DateTimeFunctions.h

```diff
@@ -298,4 +299,48 @@ struct DateSubFunction {
   }
 };
 
+template <typename T>
+struct AddMonthsFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE int getYear(const std::tm& time) {
```

Comment:
These functions seem to repeat in this file. Let's extract these into free functions in the anonymous namespace to avoid copy-paste.